src/management-system-v2/app/(dashboard)/[environmentId]/processes/[processId]/modeler-share-modal-option-embed-in-web.tsx

```diff
@@ -12,38 +12,34 @@ import { useEnvironment } from '@/components/auth-can';
 const { TextArea } = Input;
 
 type ModelerShareModalOptionEmdedInWebProps = {
-  shared: boolean;
   sharedAs: 'public' | 'protected';
   allowIframeTimestamp: number;
   refresh: () => void;
 };
 
 const ModelerShareModalOptionEmdedInWeb = ({
-  shared,
   sharedAs,
   allowIframeTimestamp,
   refresh,
 }: ModelerShareModalOptionEmdedInWebProps) => {
   const { processId } = useParams();
   const environment = useEnvironment();
-  const [isAllowEmbeddingChecked, setIsAllowEmbeddingChecked] = useState(
-    shared && sharedAs === 'public',
-  );
+  const [isAllowEmbeddingChecked, setIsAllowEmbeddingChecked] = useState(sharedAs === 'public');
```

Comment:
Is this state really necessary? At the two places where it is used it might actually be better to just check if the token is set. At the moment you see the iframe code with `...?token= height="100%" ...` if the process is publicly shared but if there is no iframeTimestamp

Reply:
I also think it is not intended, that this option shows you an error that the "Process is not shared as public" when the user selects to share the process only with logged in users but the option might also show you that embedding is activated with code ready to be copied when the process is not shared at all.
How exactly are sharing and embedding supposed to interact with each other? Isn't embedding basically sharing of public processes with some iframe code that the user could also write and one less button on the shared page?

Reply:
Did you discuss this with Kai? It seems kind of logical that embedding might be linked to sharing. The semi linked solution just seemed a bit inconsistent